### PR TITLE
Add KDBAI vector store

### PR DESCRIPTION
**Pull request by bu2kx**
Addition of KDBAI vector store (https://kdb.ai).

Dependencies: `kdbai_client` v0.1.2 Python package.

Sample notebook: `docs/docs/integrations/vectorstores/kdbai.ipynb`

Tag maintainer: @bu2kx
Twitter handle: @kxsystems

### CODE REVIEW

File: libs/langchain/langchain/vectorstores/kdbai.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+        distance_strategy: One option from DistanceStrategy.EUCLIDEAN_DISTANCE, 
+            DistanceStrategy.DOT_PRODUCT or DistanceStrategy.COSINE.
+
+    See the example [notebook](https://github.com/KxSystems/langchain/blob/KDB.AI/docs/docs/integrations/vectorstores/kdbai.ipynb).
+    """
+
+    def __init__(
+        self,
+        table: Any,
+        embedding: Union[Embeddings, Callable],
```
Comment (baskaryan) [inline libs/langchain/langchain/vectorstores/kdbai.py:36]: preference to just take Embeddings (any vecstores that take both do so for vestigial + backwards compatibility reasons)

File: libs/langchain/langchain/vectorstores/kdbai.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
+from __future__ import annotations
+
+import logging
+import uuid
+from typing import Any, Callable, Iterable, List, Optional, Tuple, Union
+
+import numpy as np
+import pandas as pd
```
Comment (baskaryan) [inline libs/langchain/langchain/vectorstores/kdbai.py:8]: pandas is optional dependency so should be imported within functions that need it

File: libs/langchain/pyproject.toml [hunk 3]
```diff
@@ -138,6 +138,7 @@ anthropic = {version = "^0.3.11", optional = true}
 aiosqlite = {version = "^0.19.0", optional = true}
 rspace_client = {version = "^2.5.0", optional = true}
 upstash-redis = {version = "^0.15.0", optional = true}
+kdbai_client = {version = "^0.1.2", optional = true}
```
Comment (baskaryan) [inline libs/langchain/pyproject.toml:141]: prefer to only include optional dependencies in pyproject if they're needed for unit testing (the all extras group is deprecated), so no need to add here

**Review by baskaryan — COMMENTED** [review]
looks awesome, few comments! happy to make changes myself if you want to give me access to the fork

File: libs/langchain/langchain/vectorstores/kdbai.py (hunk 1 above)
Comment (bu2kx) [reply libs/langchain/langchain/vectorstores/kdbai.py:36]: Thanks @baskaryan for the speedy review, I will proceed with the changes in the next couple hours.

Comment (bu2kx) [reply libs/langchain/langchain/vectorstores/kdbai.py]: Should be good following new commit:  https://github.com/langchain-ai/langchain/pull/12797/commits/ed5834c9520482f3a88d7de8983f74847906e84d

File: libs/langchain/pyproject.toml [hunk 4]
```diff
@@ -306,6 +306,7 @@ all = [
     "amadeus",
     "librosa",
     "python-arango",
+    "kdbai_client",
```
Comment (baskaryan) [inline libs/langchain/pyproject.toml:309]: you'll want to delete here, too

Comment (bu2kx) [reply libs/langchain/pyproject.toml]: Oups, done.

**Comment by bu2kx** [comment]
Dear @baskaryan, any change to get this PR merged in ? It has been passing lint check and tests locally last Friday (but the `langchain` community is adding more checks and tests faster than I can follow :-) )

**Review by hwchase17 — COMMENTED** [review]
this should now be moved to langchain-community, as that is where all integrations have moved

**Comment by bu2kx** [comment]
> this should now be moved to langchain-community, as that is where all integrations have moved

Dear @hwchase17 thanks for your feedback. 
We have rebased our integration on top of LangChain v0.1.0 relocated our vector store under `langchain_community`.
Unit tests and linting are passing successfully (locally).

**Comment by bu2kx** [comment]
Dear @baskaryan , we updated our vectorstore integration to LangChain v0.1.0 and tried to pass all tests and linting checks locally. Could you give us another CI run ? Regards

**Comment by bu2kx** [comment]
Dear @hwchase17 and @baskaryan , we updated our vectorstore integration to LangChain v0.1.0, fixed a remaining unit test (`tests/unit_tests/vectorstores/test_public_api.py::test_public_api`),  and tried to pass all tests and linting checks locally. Could you give us another CI run ? Regards

PS: Would there be an easy way to run your CI checks under our Github account ? (so I could validate everything in advance)

**Comment by bu2kx** [comment]
@baskaryan Could we get this merged in now that CI checks passed ? Regards.

File: libs/community/langchain_community/vectorstores/kdbai.py [hunk 5]
```diff
@@ -0,0 +1,238 @@
+from __future__ import annotations
+
+import logging
+import uuid
+from typing import Any, Iterable, List, Optional, Tuple
+
+import numpy as np
+
+try:
+    import pandas as pd
```
Comment (baskaryan) [inline libs/community/langchain_community/vectorstores/kdbai.py:10]: can we import this in the functions that use it, the way we import kdbai_client?

**Review by baskaryan — COMMENTED** [review]
can we remove any code changes from `libs/langchain`? only old integrations are re-exported from there for backwards compatibility reasons, all new integrations only need to touch `libs/community`. the old `libs/langchain` exports will be removed in the next minor version release of `langchain`

File: libs/community/langchain_community/vectorstores/kdbai.py (hunk 5 above)
Comment (bu2kx) [reply libs/community/langchain_community/vectorstores/kdbai.py]: I made the changes following your feedback. Thanks.

**Review by baskaryan — APPROVED** [review]
lgtm, thank you @bu2kx!

**Pull request merged**